_sources/vectors_in_cpp/mixed_up_code.rst

```diff
@@ -0,0 +1,292 @@
+        }
+
+        vector<string> exclamations = {"what?!", "how???", "fine!", "STOP.", "yay!!!!!", "ugh...!"};
+
+    -----
+    #include <vector>
+    #include <string>
+    using namespace std;
+    =====
+    int main() {
+    =====
+        vector<string> exclamations = {"what?!", "how???", "fine!", "STOP.", "yay!!!!!", "ugh...!"};
+    =====
+        vector<char> punc = {'.', '!', '?'};
+    =====
+        vector<int> counts = {};
+    =====
+        for (int i = 0; i < punc.size(); i++) {
+    =====
+            counts.push_back(howMany(excl, punc[i]));
+    =====
+        }
+    =====
+    }
+
+.. parsonsprob:: vectors_in_cpp_mixed_up_code_7
+    :numbered: left
+    :adaptive:
+
+    Write the function ``endsEven`` that takes a vector and removes elements from the end of the vector until
+    it ends with an even number. Put the necessary blocks of code in the correct order.
+    -----
+    void endsEven (vector<int> &vec) {
+    =====
+    vector endsEven (vector<int> &vec) { #distractor
+    =====
+        while (vec.back() % 2 != 0) {
+    =====
+        for (int i = 0; i < vec.size(); i++) { #paired
+    =====
+        vec.pop_back();
+    =====
+        }
+    =====
+    }
+
+.. parsonsprob:: vectors_in_cpp_mixed_up_code_8
+    :numbered: left
+    :adaptive:
+
+    Write the function ``randomNums`` that takes two integers: ``num`` which is the number of random numbers
+    you wish to generate, and ``max``, which is the maximum value of random number you wish to generate.  Your
+    function should return a vector of ``num`` integers that are between 1 and ``max``, inclusive.
+    Put the necessary blocks of code in the correct order.
+    -----
+    vector<int> randomNums (int num, int max) {
+    =====
+        vector<int> randomVec(num);
+    =====
+        for (int i = 0; i < num; i++) {
+    =====
+        for (int i = 0; i <= randomVec.size(); i++) { #paired
+    =====
+            randomVec[i] = rand() % max + 1;
+    =====
+        }
+    =====
+        return randomVec;
+    =====
+        return randomVec[i]; #distractor
+    =====
+    }
+
+.. parsonsprob:: vectors_in_cpp_mixed_up_code_9
+    :numbered: left
+    :adaptive: 
+
+    Write the function ``weird_print`` that prints the first half of a vector of integers in reverse order
+    and then prints the second half in the order present in the vector.
+    If we had ``vec = {1,2,3,4,5,6}``
+    we would print ``3 2 1 4 5 6``.
+    You can assume the size of the vector will always be even.
+    Put the necessary blocks of code in the correct order.
+    -----
+    void weird_print (vector<int> vec) {
+    =====
+        int half = vec.size() / 2;
+    =====
+        for (int i = vec.size() - 1; i >= half; i--){
+    =====
+            cout << vec[i-half] << ' ';
+    =====
+        }
+    =====
+        for (int h = 0; h < half; h++) {
+    =====
+            cout << vec[h + half] << ' ';
+    =====
+        }
+    =====
+        cout << endl;
+    =====
```

Comment:
I think this is not a very intuitive way to solve this problem. My first thought is to iterate from the middle of the vector to the beginning, printing out numbers. Then, in a second loop, iterate from the middle of the vector to the end. I think that the current solution is not the easiest way to solve this, so I would either remove this question or simplify this.